src/NumericalAlgorithms/DiscontinuousGalerkin/MetricIdentityJacobian.cpp

```diff
@@ -56,170 +61,337 @@ void metric_identity_jacobian(
+        det_jac_times_inverse_jacobian, mesh, inertial_coords, *jacobian);
+  }
+
+  // Now compute the determinant of the Jacobian, the inverse Jacobian, and the
+  // Jacobian.
+  if constexpr (Dim == 1) {
+    get(*det_jacobian) = get<0, 0>(*jacobian);
+    get<0, 0>(*inverse_jacobian) = 1.0 / get<0, 0>(*jacobian);
+  } else if constexpr (Dim == 2) {
+    // Note that det_jacobian and jacobian are used as buffers here
+    determinant_and_inverse(det_jacobian, jacobian,
+                            *det_jac_times_inverse_jacobian);
+    for (size_t storage_index = 0; storage_index < inverse_jacobian->size();
+         ++storage_index) {
+      (*inverse_jacobian)[storage_index] =
+          (*det_jac_times_inverse_jacobian)[storage_index] / get(*det_jacobian);
+    }
+    for (size_t storage_index = 0; storage_index < jacobian->size();
+         ++storage_index) {
+      (*jacobian)[storage_index] *= get(*det_jacobian);
+    }
+  } else {
+    {
+      // In 3d det(J inv_jac) = 1/J^2.
+      // We need to use a temporary (which we turn into a buffer using the
+      // inverse_jacobian allocation) in order to compute the sign correct.
+      Jacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>
+          jacobian_by_det_jacobian{};
+      for (size_t storage_index = 0;
+           storage_index < jacobian_by_det_jacobian.size(); ++storage_index) {
+        jacobian_by_det_jacobian[storage_index].set_data_ref(
+            make_not_null(&(*inverse_jacobian)[storage_index]));
+      }
+      // Note that det_jacobian is used as a buffer here
+      determinant_and_inverse(det_jacobian,
+                              make_not_null(&jacobian_by_det_jacobian),
+                              *det_jac_times_inverse_jacobian);
+
+      ASSERT(DataVector{sign(get<0, 0>(jacobian_by_det_jacobian) /
+                             get<0, 0>(*jacobian))} ==
+                     DataVector{sign(get<1, 1>(jacobian_by_det_jacobian) /
+                                     get<1, 1>(*jacobian))} and
+                 DataVector{sign(get<0, 0>(jacobian_by_det_jacobian) /
+                                 get<0, 0>(*jacobian))} ==
+                     DataVector{sign(get<2, 2>(jacobian_by_det_jacobian) /
+                                     get<2, 2>(*jacobian))},
+             "The sign of the Jacobian determinant is found to be different "
+             "depending on which component we use. The (0,0) component gives "
+                 << DataVector{sign(get<0, 0>(jacobian_by_det_jacobian) /
+                                    get<0, 0>(*jacobian))}
+                 << " the (1,1) component gives "
+                 << DataVector{sign(get<1, 1>(jacobian_by_det_jacobian) /
+                                    get<1, 1>(*jacobian))}
+                 << " and the (2,2) component gives "
+                 << DataVector{sign(get<2, 2>(jacobian_by_det_jacobian) /
+                                    get<2, 2>(*jacobian))});
```

Comment:
For my understanding: why are you only checking the diagonal components?

Reply:
That's a great question! I'm worried about round off issues in the off-diagonal components since they could be zero (~1.0e-14) but could flip flop sign conceivably. My worry is some non-trivial map that just happens to have a diagonal (or diagonally very dominant) Jacobian and therefore the off diagonal terms aren't exactly zero.

Reply:
Isn't there a concern that a map similar to DiscreteRotation could have order-unity off-diagonal components but order-zero diagonal components, making your argument valid but in the other direction?

If this is a concern, perhaps the robust (but obviously somewhat annoying) solution is to check the several largest components?

Reply:
Awe :( Yea, I think it's best to just check the largest 2/3 components. To reduce overhead think it's sufficient to use the `0` grid point to sort by?

Reply:
I don't have great intuition for this, but I'd tend to agree. Maybe with a note that about the assumptions being made so they can be revisited if needed in the future?

Reply:
Maybe the safest thing to do is to use the sum of one row or column. e.g.
```cpp
(get<0, 0>(jacobian_by_detjacobian) + get<0, 1>(jacobian_by_detjacobian) + get<0, 2>(jacobian_by_detjacobian)) / (...)
```
If an entire row/column is zero then the Jacobian is singular and we are not doing so well :P 

@kidder @fmahebert thoughts?

Reply:
Offline discussion:

The only reason the Jacobian changes sign is because maps that use discrete rotations don't actually rotate, they negate the axes, which causes the coordinate system to go from right-handed to left-handed, effectively. The right solution is to actually fix those coordinate maps and domain creators so that the sign of det(Jacobian) is preserved.